streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -69,7 +70,16 @@ public void init(ProcessorContext context) {
     @Override
     public void process(final K key, final V value) {
         RecordCollector collector = ((RecordCollector.Supplier) context).recordCollector();
-        collector.send(new ProducerRecord<>(topic, null, context.timestamp(), key, value), keySerializer, valSerializer, partitioner);
+
+        final long timestamp = context.timestamp();
+        if (timestamp < 0) {
+            throw new StreamsException("Input topic record has invalid (negative) timestamp, " +
```

Comment:
Perhaps:

> The record consumed from the input topic has an invalid (negative) timestamp, possibly because a pre-0.10 producer client was used to write this record to Kafka without embedding a timestamp, or because you are reading from a pre-0.10 topic after upgrading the Kafka cluster to 0.10+. [...]